impl/src/main/java/eu/darkbot/impl/managers/GalaxySpinner.java

```diff
@@ -49,13 +52,23 @@ public Optional<SpinResult> spinGate(@NotNull GalaxyGate gate, boolean multiplie
         if (multiplier) params += "&multiplier=1";
         if (spinAmount > 4) params += "&spinamount=" + spinAmount;
 
-        return Boolean.TRUE.equals(handleRequest(params, -1, minWait))
+        Optional<SpinResult> spinResult = Boolean.TRUE.equals(handleRequest(params, -1, minWait))
                 ? Optional.of(galaxyInfo.getSpinResult()) : Optional.empty();
+
+        if (!spinResult.isEmpty())
+            eventBroker.sendEvent(new SpinGateEvent(spinResult.get(), spinAmount));
+
+        return spinResult;
```

Comment:
```suggestion
        boolean success = Boolean.TRUE.equals(handleRequest(params, -1, minWait));
        if (success)
            eventBroker.sendEvent(new SpinGateEvent(galaxyInfo.getSpinResult(), spinAmount));

        return success ? Optional.of(galaxyInfo.getSpinResult()) : Optional.empty();
```
i'd create the optional at the very end, just as an api response, since the rest doesn't need it